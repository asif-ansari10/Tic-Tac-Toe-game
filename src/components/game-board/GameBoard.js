
import React from 'react';
import PropTypes from 'prop-types';

import GameBoardSquares from './squares/GameBoardSquares';

import './GameBoard.css';

export default class GameBoard extends React.Component {
  constructor(props) {
    super(props);
    this.squareClicked = this.squareClicked.bind(this);
    this.calculateLines = this.calculateLines.bind(this);
  }

  /* eslint-disable-next-line  */
  squareClicked(square) {
    console.log('Square clicked!', square);
  }

  calculateLines() {
    const { height, width } = this.props;
    const oneThirdHeight = height / 3;
    const twoThirdsHeight = oneThirdHeight * 2;
    const oneThirdWidth = width / 3;
    const twoThirdsWidth = oneThirdWidth * 2;

    return [
      {
        x1: oneThirdWidth, y1: 0, x2: oneThirdWidth, y2: height,
      },
      {
        x1: twoThirdsWidth, y1: 0, x2: twoThirdsWidth, y2: height,
      },
      {
        x1: 0, y1: oneThirdHeight, x2: width, y2: oneThirdHeight,
      },
      {
        x1: 0, y1: twoThirdsHeight, x2: width, y2: twoThirdsHeight,
      },
    ];
  }

  render() {
    const { height, width } = this.props;
    const lines = this.calculateLines().map(
      coords => <line key={`${coords.x1}-${coords.y1}`} {...coords} />,
    );

    return (
      <svg height={height} width={width}>
        {lines}
        <GameBoardSquares
          height={height}
          width={width}
          board={[0, 0, 0, 1, 1, 1, 2, 2, 2]}
        />
      </svg>
    );
  }
}

GameBoard.propTypes = {
  height: PropTypes.number,
  width: PropTypes.number,
};

GameBoard.defaultProps = {
  height: 500,
  width: 500,
};
