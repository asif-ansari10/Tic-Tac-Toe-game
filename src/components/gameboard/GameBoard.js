
import React from 'react';

export default class GameBoard extends React.Component {
  constructor(props) {
    super(props);
    this.squareClicked = this.squareClicked.bind(this);
  }

  squareClicked(square) {
    this.console.log('Square clicked!', square);
  }

  render() {
    return (
      <span>
        Hello!
      </span>
    );
  }
}
