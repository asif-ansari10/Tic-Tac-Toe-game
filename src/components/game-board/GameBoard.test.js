/* global beforeEach, describe, expect, jest, it */
import React from 'react';
import { shallow } from 'enzyme';

import GameBoard from './GameBoard';

beforeEach(() => {
  jest.clearAllMocks();
});

describe('Calculated Lines', () => {
  const wrapper = shallow(<GameBoard height={500} width={500} />);
  const lines = wrapper.instance().calculateLines();

  it('should have correct first vertical line', () => {
    expect(lines[0]).toEqual({
      x1: (500 / 3),
      y1: 0,
      x2: (500 / 3),
      y2: 500,
    });
  });

  it('should have correct second vertical line', () => {
    expect(lines[1]).toEqual({
      x1: (500 / 3) * 2,
      y1: 0,
      x2: (500 / 3) * 2,
      y2: 500,
    });
  });

  it('should have correct first horizontal line', () => {
    expect(lines[2]).toEqual({
      x1: 0,
      y1: 500 / 3,
      x2: 500,
      y2: 500 / 3,
    });
  });

  it('should have correct second horizontal line', () => {
    expect(lines[3]).toEqual({
      x1: 0,
      y1: (500 / 3) * 2,
      x2: 500,
      y2: (500 / 3) * 2,
    });
  });
});

it('should match a snapshot', () => {
  const wrapper = shallow(<GameBoard />);
  expect(wrapper).toMatchSnapshot();
});
