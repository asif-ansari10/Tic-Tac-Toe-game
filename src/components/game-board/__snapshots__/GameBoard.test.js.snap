// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameBoard
    height={500}
    width={500}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <line
            x1={166.66666666666666}
            x2={166.66666666666666}
            y1={0}
            y2={500}
/>,
          <line
            x1={333.3333333333333}
            x2={333.3333333333333}
            y1={0}
            y2={500}
/>,
          <line
            x1={0}
            x2={500}
            y1={166.66666666666666}
            y2={166.66666666666666}
/>,
          <line
            x1={0}
            x2={500}
            y1={333.3333333333333}
            y2={333.3333333333333}
/>,
        ],
        <GameBoardSquares
          board={
                    Array [
                              0,
                              0,
                              0,
                              1,
                              1,
                              1,
                              2,
                              2,
                              2,
                            ]
          }
          height={500}
          width={500}
/>,
      ],
      "height": 500,
      "width": 500,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "166.66666666666666-0",
        "nodeType": "host",
        "props": Object {
          "x1": 166.66666666666666,
          "x2": 166.66666666666666,
          "y1": 0,
          "y2": 500,
        },
        "ref": null,
        "rendered": null,
        "type": "line",
      },
      Object {
        "instance": null,
        "key": "333.3333333333333-0",
        "nodeType": "host",
        "props": Object {
          "x1": 333.3333333333333,
          "x2": 333.3333333333333,
          "y1": 0,
          "y2": 500,
        },
        "ref": null,
        "rendered": null,
        "type": "line",
      },
      Object {
        "instance": null,
        "key": "0-166.66666666666666",
        "nodeType": "host",
        "props": Object {
          "x1": 0,
          "x2": 500,
          "y1": 166.66666666666666,
          "y2": 166.66666666666666,
        },
        "ref": null,
        "rendered": null,
        "type": "line",
      },
      Object {
        "instance": null,
        "key": "0-333.3333333333333",
        "nodeType": "host",
        "props": Object {
          "x1": 0,
          "x2": 500,
          "y1": 333.3333333333333,
          "y2": 333.3333333333333,
        },
        "ref": null,
        "rendered": null,
        "type": "line",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "board": Array [
            0,
            0,
            0,
            1,
            1,
            1,
            2,
            2,
            2,
          ],
          "height": 500,
          "width": 500,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "svg",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <line
              x1={166.66666666666666}
              x2={166.66666666666666}
              y1={0}
              y2={500}
/>,
            <line
              x1={333.3333333333333}
              x2={333.3333333333333}
              y1={0}
              y2={500}
/>,
            <line
              x1={0}
              x2={500}
              y1={166.66666666666666}
              y2={166.66666666666666}
/>,
            <line
              x1={0}
              x2={500}
              y1={333.3333333333333}
              y2={333.3333333333333}
/>,
          ],
          <GameBoardSquares
            board={
                        Array [
                                    0,
                                    0,
                                    0,
                                    1,
                                    1,
                                    1,
                                    2,
                                    2,
                                    2,
                                  ]
            }
            height={500}
            width={500}
/>,
        ],
        "height": 500,
        "width": 500,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "166.66666666666666-0",
          "nodeType": "host",
          "props": Object {
            "x1": 166.66666666666666,
            "x2": 166.66666666666666,
            "y1": 0,
            "y2": 500,
          },
          "ref": null,
          "rendered": null,
          "type": "line",
        },
        Object {
          "instance": null,
          "key": "333.3333333333333-0",
          "nodeType": "host",
          "props": Object {
            "x1": 333.3333333333333,
            "x2": 333.3333333333333,
            "y1": 0,
            "y2": 500,
          },
          "ref": null,
          "rendered": null,
          "type": "line",
        },
        Object {
          "instance": null,
          "key": "0-166.66666666666666",
          "nodeType": "host",
          "props": Object {
            "x1": 0,
            "x2": 500,
            "y1": 166.66666666666666,
            "y2": 166.66666666666666,
          },
          "ref": null,
          "rendered": null,
          "type": "line",
        },
        Object {
          "instance": null,
          "key": "0-333.3333333333333",
          "nodeType": "host",
          "props": Object {
            "x1": 0,
            "x2": 500,
            "y1": 333.3333333333333,
            "y2": 333.3333333333333,
          },
          "ref": null,
          "rendered": null,
          "type": "line",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "board": Array [
              0,
              0,
              0,
              1,
              1,
              1,
              2,
              2,
              2,
            ],
            "height": 500,
            "width": 500,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "svg",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
